services/data-submission.js

```diff
@@ -0,0 +1,131 @@
+        // Default conditons are:
+        // Make sure application has valid status
+        let conditions = [{$and: [validApplicationStatus]}];
+        
+        // Filter on organization and status
+        if (params.organization !== "All") {
+            conditions[0].$and.push({"organization.name": params.organization});
+        }
+        if (params.status !== "All") {
+            conditions[0].$and.push({status: params.status});
+        }
+        // List all applications if Fed Lead / Admin / Data Concierge / Data Curator
+        const listAllApplicationRoles = [USER.ROLES.ADMIN, USER.ROLES.FEDERAL_LEAD, USER.ROLES.CURATOR, USER.ROLES.DC_POC];
+        if (listAllApplicationRoles.includes(userRole)) {
+            return [{"$match": {"$or": conditions}}];;
+        }
+        // If org owner, add condition to return all data submissions associated with their organization
+        if (userRole === USER.ROLES.ORG_OWNER && aUserOrganization?.orgID) {
+            conditions[0].$and.push({"organization.name": aUserOrganization.orgName});
+            return [{"$match": {"$or": conditions}}];;
+        }
+
+        // Add condition so submitters will only see their data submissions
+        // User's cant make submissions, so they will always have no submissions 
+        // search by applicant's user id
+        conditions[0].$and.push({"submitterID": userID});
+
+        return [{"$match": {"$or": conditions}}];
+    }
+
+    async listDataSubmissions(params, context) {
+        verifySession(context)
+            .verifyInitialized();
+        let pipeline = this.listConditions(context.userInfo._id, context.userInfo?.role, context.userInfo?.organization, params);
+        // let pipeline = [];
+        if (params.orderBy) pipeline.push({"$sort": { [params.orderBy]: getSortDirection(params.sortDirection) } });
+
+        const pagination = [];
+        if (params.offset) pagination.push({"$skip": params.offset});
+        const disablePagination = Number.isInteger(params.first) && params.first === -1;
+        if (!disablePagination) {
+            pagination.push({"$limit": params.first});
+        }
+        const promises = [
+            await this.dataSubmissionCollection.aggregate((!disablePagination) ? pipeline.concat(pagination) : pipeline),
+            await this.dataSubmissionCollection.aggregate(pipeline)
+        ];
+        
+        return await Promise.all(promises).then(function(results) {
+            return {
+                submissions: results[0] || [],
+                total: results[1]?.length || 0
+            }
+        });
+    }
+
+
+}
+
+function formatApplicantName(userInfo){
+    if (!userInfo) return "";
+    let firstName = userInfo?.firstName || "";
+    let lastName = userInfo?.lastName || "";
+    lastName = lastName.trim();
+    return firstName + (lastName.length > 0 ? " "+lastName : "");
+}
```

Comment:
This is duplicates codes in application.js. It should create a public function if possible.